From 432e5a6f3e8be2ed369e8e7acec032358b750887 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 10 Nov 2022 10:09:58 +0200
Subject: [PATCH] [SW-81037] gaudi3: pif: don't load d1 pif

It was decided that d2d phy will not be initialized in ppboot,
and there is no need to do FLR reset in ppboot. So there is no
need to load the pif arc of D1 in ppboot.

Change-Id: I724c802e1e1ea241223bbf593611bc3a98b36056
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index d2ee94b..cc52c61 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -712,11 +712,6 @@
 	ppboot_run_pif_arc(data);
 }
 
-static bool hl_is_slave_pif_isolated(void)
-{
-	return !!regmap_readl(mmD1_PSOC_BOOT_CONF_D1_PIF_ISO);
-}
-
 void preboot_pcie_wait_for_link(void)
 {
 	struct cpu_load_data load_data = {
@@ -728,18 +723,4 @@
 	};
 
 	ppboot_load_and_run_pif_arc(&load_data);
-
-	if (!hl_is_dual_die_sku())
-		return;
-
-	if (hl_is_slave_pif_isolated())
-		return;
-
-	load_data.cpu_cfg_base = mmD1_PIF_ARC_BASE;
-	load_data.img_type[PIF_ARC_ICCM_IDX] = HL_FW_IMG_D1_PIF_ICCM;
-	load_data.img_type[PIF_ARC_DCCM_IDX] = HL_FW_IMG_D1_PIF_DCCM;
-	load_data.die_id = GAUDI3_SLAVE_DIE_ID;
-	ppboot_load_and_run_pif_arc(&load_data);
-
-	/* TODO: if slave pif is isolated, load and run d1 psoc arc (SW-98975) */
 }
